lib/openhab/core/types/quantity_type.rb

```diff
@@ -274,6 +294,26 @@ def #{ruby_op}(other)
           )
         end
 
+        #
+        # @return [Duration] a {Duration} if the unit is time-based
+        # @return [nil] if the unit is not time-based
```

Comment:
why return nil instead of raise if it's not compatible? as far as I'm concerned, this method is unnecessary - just use `to_temporal_amount`

Reply:
Don't you think it's nicer to be able to check without creating a rescue block?